### DISCUSSION TITLE
newbie: I want to create a switchboard with 30 switches and read the status of it

### DISCUSSION THREAD

**Discussion by bartplessers** [arendst/Tasmota#18835] [Q&A]
Hi everyone,

I recently aquired an old electric board with 30 (depricated) automatic fuses:
![IMG_0874](https://github.com/arendst/Tasmota/assets/10470708/921092d4-6561-431d-bbef-6429a4eb3f91)

Just for fun: I want to use those fuses as "sensors" by integrating their status (open/closed)  in my Home Assistant. 
So I want to read the “state” of a fuse: on/off. 
Main idea: 
- use an ESP8266 or ESP32
- I already use Tasmota for quite some devices, so maybe I can extend this firmware for my purpose
- I want to have 30 physical attached control leds
- I want to get the status of each fuse in Home Assistant

Can this be done with an ESP8266/ESP32 and Tasmota?

This is all very new to me as -till now- I was only "consuming" Tasmota on some sonoff devices with out-of-the box functionality...
So any kind help in baby steps is appreciated :-)

kind regards,
Bart Plessers

**Comment by sfromis** [comment]
If the leds are to show the switch status, you could wire it to work that way, lessening the number of pins needed. As Tasmota only allows up to 28 switch inputs, simplest may be two ESP32 boards, 15 switches on each. If leds needs separate pins, that could then be 15 relay outputs, and make sure to get a board with at least 30 gpios broken out.

It would also be doable to treat the 30 switches as a keyboard, in a 6x6 matrix, and have some code to detect which are on, by pulsing 6 lines, and see where signals are returned. I don't remember seeing this as a Tasmota feature, but haven't looked at all devices.

**Comment by pkkrusty** [comment]
Using a multiplexer could work too, polling the status of the switches progressively using scripting. Could be done with a single ESP32 probably, since the multiplexer uses 5 pins to read up to 16 inputs. See the scripting example for the flower garden.

**Reply by barbudor** [reply]
Shift registers would be the best and there are already support in Tasmota for boards that support 32 inputs/32 outputs such as the Dingtians
You cans also mimic (to avoid saying copying) the design if you want to use the existing driver with a board you would design yourself

**Reply by barbudor** [reply]
Or i2c i/o expanders such as PCF8574 and MCP230xx
